### PR TITLE
Enable Repolinter checks on main

### PR DESCRIPTION
**Pull request by mynameistechno**
Adds [Repolinter](https://github.com/todogroup/repolinter) checks to PRs and pushes to main. Repoliner is a TODO Group (Linux Foundation) created project that lints for common open source repo issues. 

The [config](https://github.com/quic/.github/blob/main/repolint.json) is pulled from our main QuIC org, but if we need to override we can [extend and configure it locally](https://github.com/quic/.github/tree/main?tab=readme-ov-file#customizing-repolinter-rules).

Let me know if any concerns with enabling this check on this repo. Also, should we enable for other repos such as qcom distro?

### CODE REVIEW

**Comment by mynameistechno** [comment]
Looking at this ...

image

**Comment by mynameistechno** [comment]
> Looking at this ...
> 
> image

Ok, I had to add that action to the allowed list

File: ci/yocto-check-layer.sh [hunk 1]
```diff
@@ -1,4 +1,6 @@
 #!/bin/sh -e
+# Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: MIT
```
Comment (craigez) [inline ci/yocto-check-layer.sh:3]: We can exclude such trivial shell scripts, but it's probably easier to add the markings.

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**